scipy/spatial/src/distance_metrics.h

```diff
@@ -449,7 +449,7 @@ struct JaccardDistance {
     void operator()(StridedView2D<T> out, StridedView2D<const T> x, StridedView2D<const T> y) const {
         transform_reduce_2d_<4>(out, x, y, [](T x, T y) INLINE_LAMBDA {
             Acc<T> acc;
-            acc.num = (x != y) & ((x != 0) | (y != 0));
+            acc.num = (x != 0) ^ (y != 0);
```

Comment:
the changes in this file have no impact on the Python function, right? If so, could you put them in a separate commit?

Reply:
Correct, the C++ change and the Python change are independent.

Let me try rearrange the commits to "one at a time".